STM32N6: XSPI PSRAM: Configure prescaler value automatically

gautierg-st opened the pull request:
Add the automatic computation / configuration of the XSPI prescaler for the XSPI PSRAM driver.

gautierg-st committed:
dts: bindings: memc: add a max-frequency property for stm32 xspi psram

Add a parameter to define the maximum PSRAM frequency for the STM32 XSPI
driver. It will be useful to automatically calculate a prescaler.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
drivers: memc: compute prescaler automatically for stm32 xspi

For STM32 XSPI PSRAM driver, compute and set the prescaler automatically
according to the kernel clock and the max frequency of the PSRAM.
Copied from what is done in the STM32 XSPI Flash driver.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

gautierg-st committed:
boards: st: stm32n6570_dk: add xspi psram max frequency

Add XSPI PSRAM maximum frequency property in the dts.

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

mathieuchopstm commented on drivers/memc/memc_stm32_xspi_psram.c:
```diff
@@ -259,6 +265,17 @@ static int memc_stm32_xspi_psram_init(const struct device *dev)
+			break;
+		}
+	}
+	__ASSERT_NO_MSG(prescaler >= STM32_XSPI_CLOCK_PRESCALER_MIN &&
+			prescaler <= STM32_XSPI_CLOCK_PRESCALER_MAX);
```
Nit:
```suggestion
	__ASSERT_NO_MSG(IN_RANGE(prescaler, STM32_XSPI_CLOCK_PRESCALER_MIN, STM32_XSPI_CLOCK_PRESCALER_MAX));
```

gautierg-st replied:
Done

gautierg-st committed:
drivers: flash: stm32 xspi: replace prescaler assert by returning error

Replace the assert on the value of the prescaler by a standard check and
an error value return (similar to the STM32 XSPI PSRAM driver).

Signed-off-by: Guillaume Gautier <guillaume.gautier-ext@st.com>

erwango reviewed (dismissed):
Approved but let's wait for https://github.com/zephyrproject-rtos/zephyr/pull/88646 to be merged first

gautierg-st commented:
Rebased on top of FRASTM PR #88646
